Could not create unix socket lock file error in mysql service when timezones are imported

If `importTimeZones` is true or `default-time-zone` then the `configureTimezones` script will be executed during `startScript` initialization, if and only the dataDir does not exists.

The initialization fails with similar error:
```text
[mysql	] 2024-04-22T15:42:53.257499Z 0 [ERROR] Could not create unix socket lock file ./data/mysql/config.sock.lock.
```

The socket path used in `configureTimezones` script is relative and mysql fails to create it.

--- a/nix/mysql/default.nix
+++ b/nix/mysql/default.nix
@@ -193,16 +193,15 @@ in
             configureTimezones = ''
               # Start a temp database with the default-time-zone to import tz data
               # and hide the temp database from the configureScript by setting a custom socket
-              nohup mysqld ${mysqldOptions} --socket="${config.dataDir}/config.sock" --skip-networking --default-time-zone=SYSTEM &
+              CONFIG_SOCKET="$(${pkgs.coreutils}/bin/realpath ${config.dataDir + "/config.sock"})"
+              nohup mysqld ${mysqldOptions} --socket="$CONFIG_SOCKET" --skip-networking --default-time-zone=SYSTEM &
 
-              while ! MYSQL_PWD="" mysqladmin --socket="${config.dataDir}/config.sock" ping -u root --silent; do
+              while ! MYSQL_PWD="" mysqladmin --socket="$CONFIG_SOCKET" ping -u root --silent; do
                 sleep 1
               done
-
-              mysql_tzinfo_to_sql ${pkgs.tzdata}/share/zoneinfo/ | MYSQL_PWD="" mysql --socket="${config.dataDir}/config.sock" -u root mysql
-
+              mysql_tzinfo_to_sql ${pkgs.tzdata}/share/zoneinfo | MYSQL_PWD="" mysql --socket="$CONFIG_SOCKET" -u root mysql
               # Shutdown the temp database
-              MYSQL_PWD="" mysqladmin --socket="${config.dataDir}/config.sock" shutdown -u root
+              MYSQL_PWD="" mysqladmin --socket="$CONFIG_SOCKET" shutdown -u root
             '';
 
             startScript = pkgs.writeShellApplication {

--- a/nix/mysql/mysql_test.nix
+++ b/nix/mysql/mysql_test.nix
@@ -7,6 +7,12 @@
       CREATE USER bar;
     '';
   };
+  services.mysql.m3 = {
+    enable = true;
+    importTimeZones = true;
+    package = pkgs.mysql80;
+    settings.mysqld.port = 3309;
+  };
   services.mysql.m1.ensureUsers = [
     {
       name = "test_database";
@@ -14,8 +20,11 @@
       ensurePermissions = { "test_database.*" = "ALL PRIVILEGES"; };
     }
   ];
-  services.mysql.m2.enable = true;
-  services.mysql.m2.settings.mysqld.port = 3308;
+  services.mysql.m2 = {
+    enable = true;
+    importTimeZones = true;
+    settings.mysqld.port = 3308;
+  };
   settings.processes.test =
     {
       command = pkgs.writeShellApplication {
@@ -28,12 +37,29 @@
           fi
           echo 'SELECT VERSION();' | MYSQL_PWD="" mysql -h 127.0.0.1
           echo 'SELECT VERSION();' | MYSQL_PWD="" mysql -h 127.0.0.1 -P 3308
+          echo 'SELECT VERSION();' | MYSQL_PWD="" mysql -h 127.0.0.1 -P 3309 -u root
 
           echo "Checking if users foo and bar are present: "
           isFooPresent=$(echo "SELECT EXISTS(SELECT 1 FROM mysql.user WHERE user = 'foo');" | MYSQL_PWD="" mysql -h 127.0.0.1 -u root | tail -n1)
           isBarPresent=$(echo "SELECT EXISTS(SELECT 1 FROM mysql.user WHERE user = 'bar');" | MYSQL_PWD="" mysql -h 127.0.0.1 -u root | tail -n1)
           echo "$isFooPresent" | grep 1
           echo "$isBarPresent" | grep 1
+
+          echo "Checking whether named time zones are available: "
+          tz_names=$(echo 'SELECT COUNT(*) AS tz_names FROM mysql.time_zone_name;' | MYSQL_PWD="" mysql -h 127.0.0.1 -P 3309 -u root | tail -n1)
+          if [[ "$tz_names" -eq 0 ]]; then
+            echo "time_zone_name table is not populated on mysql"
+            exit 1
+          else
+            echo "time_zone_name table is correctly populated with $tz_names rows n mysql"
+          fi
+          tz_names=$(echo 'SELECT COUNT(*) AS tz_names FROM mysql.time_zone_name;' | MYSQL_PWD="" mysql -h 127.0.0.1 -P 3308 -u root | tail -n1)
+          if [[ "$tz_names" -eq 0 ]]; then
+            echo "time_zone_name table is not populated on mariadb"
+            exit 1
+          else
+            echo "time_zone_name table is correctly populated with $tz_names rows on mariadb"
+          fi
 
           echo "Checking if both foo.sql and bar.sql are executed, ignoring baz.md"
           echo "SELECT * FROM information_schema.tables WHERE table_schema = 'test_database' AND table_name = 'foo' LIMIT 1;" | MYSQL_PWD="" mysql -h 127.0.0.1 -u root | grep foo
@@ -50,6 +76,7 @@
       depends_on = {
         m1-configure.condition = "process_completed";
         m2-configure.condition = "process_completed";
+        m3-configure.condition = "process_completed";
       };
     };
 }

